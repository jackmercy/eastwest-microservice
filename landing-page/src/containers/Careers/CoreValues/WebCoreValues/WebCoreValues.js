import React, { memo, useCallback, useMemo, useRef, useState } from 'react';
import PropTypes from 'prop-types';
import Slider from 'react-slick';
import classNames from 'classnames';
import { useIntl } from 'react-intl';

import { SectionTitle } from '../../components';

import useStyles from './webCoreValues.styles';

const WebCoreValues = ({ data }) => {
  const slideRef = useRef();
  const intl = useIntl();
  const classes = useStyles();
  const settings = useMemo(
    () => ({
      dots: false,
      infinite: true,
      slidesToShow: 1,
      slidesToScroll: 1,
      arrows: false,
      autoplay: true,
      pauseOnHover: true,
      touchMove: false,
      autoplaySpeed: 10000,
    }),
    [],
  );
  const [activatedIndex, setActiveIndex] = useState(0);

  const handleCoreValueChange = useCallback(
    (index) => {
      if (slideRef.current) {
        slideRef.current.slickGoTo(index);
      }
    },
    [slideRef],
  );

  return (
    <>
      <Slider
        ref={slideRef}
        className={classes.slider}
        beforeChange={(current, next) => {
          setActiveIndex(next);
        }}
        initialSlide={activatedIndex}
        {...settings}
      >
        {data.map((item) => (
          <div key={item.key} className={classes.detailsWrapper}>
            <div className={classes.details}>
              <p className={classes.detailsSubTitle}>
                {intl.formatMessage({ id: 'CAREERS.MENU.02' })}
              </p>
              <SectionTitle
                id={`core-value-${item.key}`}
                translateId={item.titleTranslationId}
              />
              <p className={classes.detailsDescription}>{item.description}</p>
            </div>
            <img
              alt={item.title}
              src={item.image}
              className={classes.image}
              srcSet={item.srcSet}
            />
          </div>
        ))}
      </Slider>
      <div className={classes.listKeys}>
        {data.map((item, index) => (
          <div
            role="presentation"
            key={`core-value-${item.key}`}
            className={classNames(
              classes.listKeysItem,
              activatedIndex === index && 'active',
            )}
            onClick={() => {
              handleCoreValueChange(index);
            }}
          >
            <p className={classes.listKeysNumber}>{`0${index + 1}`}</p>
            <h2 className={classes.listKeysLabel}>{item.title}</h2>
          </div>
        ))}
      </div>
    </>
  );
};
WebCoreValues.propTypes = {
  data: PropTypes.arrayOf(
    PropTypes.shape({
      key: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      titleTranslationId: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
      image: PropTypes.string.isRequired,
      srcSet: PropTypes.string.isRequired,
    }),
  ).isRequired,
};

export default memo(WebCoreValues);
