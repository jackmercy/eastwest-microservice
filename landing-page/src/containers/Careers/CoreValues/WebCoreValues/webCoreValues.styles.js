import { createUseStyles } from 'react-jss';

import { rem } from '../../../../utils/jss';
import COLORS from '../../../../constants/COLORS';

export default createUseStyles({
  slider: {
    '& .slick-track, & .slick-slide': {
      display: 'flex',
      height: 'auto',
      alignItems: 'center',
      justifyContent: 'center',
    },
    '& .slick-dots': { left: 0 },
    '& .slick-prev': { left: 0 },
    '& .slick-next': { right: 0 },
    '& .slick-slide > div': { width: '100%' },
  },
  detailsWrapper: {
    display: 'inline-flex !important',
    boxSizing: 'border-box',
    alignItems: 'flex-start',
    justifyContent: 'space-between',
    padding: [0, rem(4)],
    '&:focus': { outline: 'none' },
  },
  details: { width: '100%', maxWidth: rem(528) },
  detailsSubTitle: {
    margin: 0,
    fontWeight: 600,
    fontSize: rem(12),
    lineHeight: rem(16),
    color: COLORS.primary,
  },
  detailsDescription: {
    fontSize: rem(24),
    lineHeight: rem(40),
    marginTop: rem(24),
    marginBottom: 0,
  },
  image: {
    display: 'block',
    boxSizing: 'border-box',
    marginLeft: '5%',
    flex: 'none',
    maxHeight: rem(426),
    minWidth: rem(280),
    maxWidth: rem(376),
  },
  listKeys: {
    display: 'flex',
    justifyContent: 'space-between',
    marginTop: rem(84),
  },
  listKeysItem: {
    display: 'flex',
    flex: 1,
    boxSizing: 'border-box',
    cursor: 'pointer',
    '&:nth-child(2), &:nth-child(3)': { padding: [0, rem(8)] },
    '&.active': { color: COLORS.primary },
  },
  listKeysNumber: {
    marginTop: rem(-5),
    marginBottom: 0,
    fontWeight: 900,
    fontSize: rem(32),
    lineHeight: rem(40),
  },
  listKeysLabel: {
    margin: [0, 0, 0, rem(20)],
    fontWeight: 600,
    fontSize: rem(24),
    lineHeight: rem(32),
  },
});
