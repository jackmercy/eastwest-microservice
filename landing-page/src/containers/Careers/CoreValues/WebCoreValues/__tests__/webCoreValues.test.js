import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Slider from 'react-slick';

import WebCoreValues from '../WebCoreValues';

describe('Testing Careers > CoreValues > WebCoreValues', () => {
  let wrapper;
  const props = {
    data: [
      {
        key: 'key-01',
        title: 'title 01',
        titleTranslationId: 'CAREERS.CORE_VALUE.01.TITLE',
        description: 'descriptipn 01',
        image: 'image-01.png',
        srcSet: 'image-01.png 500w',
      },
      {
        key: 'key-02',
        title: 'title 01',
        titleTranslationId: 'CAREERS.CORE_VALUE.02.TITLE',
        description: 'description 02',
        image: 'image-02.png',
        srcSet: 'image-02.png 500w',
      },
    ],
  };

  beforeAll(() => {
    wrapper = mount(<WebCoreValues {...props} />);
  });

  it('should render match snapshot', () => {
    const wrapper = shallow(<WebCoreValues {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have a Slider', () => {
    expect(wrapper.find(Slider)).toHaveLength(1);
  });
});
