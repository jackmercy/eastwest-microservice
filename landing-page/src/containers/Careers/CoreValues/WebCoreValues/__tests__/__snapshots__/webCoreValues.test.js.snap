// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Careers > CoreValues > WebCoreValues should render match snapshot 1`] = `
<Fragment>
  <Slider
    arrows={false}
    autoplay={true}
    autoplaySpeed={10000}
    beforeChange={[Function]}
    className="slider-0-2-1"
    dots={false}
    infinite={true}
    initialSlide={0}
    pauseOnHover={true}
    slidesToScroll={1}
    slidesToShow={1}
    touchMove={false}
  >
    <div
      className="detailsWrapper-0-2-2"
      key="key-01"
    >
      <div
        className="details-0-2-3"
      >
        <p
          className="detailsSubTitle-0-2-4"
        >
          Core values
        </p>
        <Memo(SectionTitle)
          id="core-value-key-01"
          translateId="CAREERS.CORE_VALUE.01.TITLE"
        />
        <p
          className="detailsDescription-0-2-5"
        >
          descriptipn 01
        </p>
      </div>
      <img
        alt="title 01"
        className="image-0-2-6"
        src="image-01.png"
        srcSet="image-01.png 500w"
      />
    </div>
    <div
      className="detailsWrapper-0-2-2"
      key="key-02"
    >
      <div
        className="details-0-2-3"
      >
        <p
          className="detailsSubTitle-0-2-4"
        >
          Core values
        </p>
        <Memo(SectionTitle)
          id="core-value-key-02"
          translateId="CAREERS.CORE_VALUE.02.TITLE"
        />
        <p
          className="detailsDescription-0-2-5"
        >
          description 02
        </p>
      </div>
      <img
        alt="title 01"
        className="image-0-2-6"
        src="image-02.png"
        srcSet="image-02.png 500w"
      />
    </div>
  </Slider>
  <div
    className="listKeys-0-2-7"
  >
    <div
      className="listKeysItem-0-2-8 active"
      key="core-value-key-01"
      onClick={[Function]}
      role="presentation"
    >
      <p
        className="listKeysNumber-0-2-9"
      >
        01
      </p>
      <h2
        className="listKeysLabel-0-2-10"
      >
        title 01
      </h2>
    </div>
    <div
      className="listKeysItem-0-2-8"
      key="core-value-key-02"
      onClick={[Function]}
      role="presentation"
    >
      <p
        className="listKeysNumber-0-2-9"
      >
        02
      </p>
      <h2
        className="listKeysLabel-0-2-10"
      >
        title 01
      </h2>
    </div>
  </div>
</Fragment>
`;
